tests/unit_tests/serialization.cpp

```diff
@@ -616,6 +616,50 @@ TEST(Serialization, serializes_ringct_types)
   ASSERT_EQ(bp0, bp1);
 }
 
+TEST(Serialization, key_encryption_transition)
+{
+  const cryptonote::network_type nettype = cryptonote::TESTNET;
+  tools::wallet2 w(nettype);
+  const boost::filesystem::path wallet_file = unit_test::data_dir / "wallet_9svHk1";
+  const boost::filesystem::path key_file = unit_test::data_dir / "wallet_9svHk1.keys";
+  const boost::filesystem::path new_wallet_file = unit_test::data_dir / "wallet_9svHk1_new";
+  const boost::filesystem::path new_key_file = unit_test::data_dir / "wallet_9svHk1_new.keys";
+  string password = "test";
+  bool r = false;
+  try
+  {
+    // Copy the old files before we overwrite them for this test
+    boost::filesystem::copy(wallet_file,new_wallet_file);
+    boost::filesystem::copy(key_file,new_key_file);
+
+    // Key transition
+    w.load(wallet_file.string(), password); // old key method
+    ASSERT_TRUE(w.get_load_info().is_legacy_key_encryption);
+    const crypto::secret_key view_secret_key = w.get_account().get_keys().m_view_secret_key;
+
+    w.rewrite(wallet_file.string(), password); // transition to new key format
+
+    w.load(wallet_file.string(), password); // new key method
+    ASSERT_FALSE(w.get_load_info().is_legacy_key_encryption);
+    ASSERT_EQ(w.get_account().get_keys().m_view_secret_key,view_secret_key);
+
+    // Restore the old files
+    MINFO("Restoring wallet file...");
```

Comment:
Why not just use `rename`?

Reply:
For some reason I thought there was could be issues using `rename` that depended on source and destination file sizes if the destination already existed. But apparently this isn't the case.